sphinx_proof/nodes.py

```diff
@@ -64,8 +87,49 @@ def depart_proof_node(self, node: Node) -> None:
     pass
 
 
+def find_parent(env, node, parent_tag):
+    """Find the nearest parent node with the given tagname."""
+    while True:
+        node = node.parent
+        if node is None:
+            return None
+        # parent should be a document in toc
+        if (
+            "docname" in node.attributes
+            and env.titles[node.attributes["docname"]].astext().lower()
+            in node.attributes["names"]
+        ):
+            return node.attributes["docname"]
+
+    if node.tagname == parent_tag:
+        return node.attributes["docname"]
+
+    return None
+
+
 def get_node_number(self: HTMLTranslator, node: Node) -> str:
+    """Get the number for the directive node for HTML."""
     key = "proof"
     ids = node.attributes.get("ids", [])[0]
     number = self.builder.fignumbers.get(key, {}).get(ids, ())
     return ".".join(map(str, number))
+
+
+def get_node_number_latex(self: LaTeXTranslator, node: Node) -> str:
+    """Get the number for the directive node for LaTeX."""
+    key = "proof"
+    docname = find_parent(self.builder.env, node, "section")
+    ids = node.attributes.get("ids", [])[0]
+    fignumbers = self.builder.env.toc_fignumbers.get(
+        docname, {}
+    )  # Latex does not have builder.fignumbers
```

Comment:
@AakashGfude I guess this is why. Given this is the only line perhaps we can have one function and then test for `isinstance`? As these functions do the same work. Just an idea.

Reply:
Good idea @mmcky . Will do it.